src/main/java/nl/tudelft/scrumbledore/keybinding/KeybindingContainer.java

```diff
@@ -0,0 +1,77 @@
+package nl.tudelft.scrumbledore.keybinding;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javafx.scene.input.KeyCode;
+import nl.tudelft.scrumbledore.Constants;
+import nl.tudelft.scrumbledore.level.PlayerAction;
+
+/**
+ * A custom container for Keybindings. One Keybinding per player.
+ * 
+ * @author Jeroen Meijer
+ */
+public final class KeybindingContainer {
+  private static KeybindingContainer keybindingContainer;
+
+  private List<Keybinding> keybindingList;
+
+  private KeybindingContainer() {
+    keybindingList = new ArrayList<Keybinding>();
+    reset();
+  }
+
+  /**
+   * Gets the KeybindingContainer.
+   * 
+   * @return KeybindingContainer the only instance.
+   */
+  public static synchronized KeybindingContainer getInstance() {
```

Comment:
Maybe update this method to use the double locking thread safety method, as implemented in the Logger class?
<organisation